src/squidpy/gr/_build.py

```diff
@@ -109,7 +144,54 @@ def spatial_neighbors(
         - :attr:`anndata.AnnData.uns`  ``['{{key_added}}']`` - :class:`dict` containing parameters.
     """
     if isinstance(adata, SpatialData):
-        adata = adata.table
+        assert (
+            elements_to_coordinate_systems is not None
+        ), "Since `adata` is a :class:`spatialdata.SpatialData`, `elements_to_coordinate_systems` must not be `None`."
```

Comment:
Why? One could pass a `SpatialData` object to calculate a standard neighbor graph, right?

Reply:
Now that we support multiple tables it could be that `sdata['table']` (`sdata.table` is deprecated), doesn't exist anymore. I think it's better to have the decoupled logic between `SpatialData` and `AnnData` as Giovanni implemented.

Reply:
yes, now we need to specify which table to use.